app/scripts/modules/openstack/loadBalancer/configure/wizard/upsert.controller.js

```diff
@@ -0,0 +1,204 @@
+      }
+    }
+
+    function onTaskComplete() {
+      application.loadBalancers.refresh();
+      application.loadBalancers.onNextRefresh($scope, onApplicationRefresh);
+    }
+
+    $scope.taskMonitor = taskMonitorService.buildTaskMonitor({
+      application: application,
+      title: (isNew ? 'Creating ' : 'Updating ') + 'your load balancer',
+      modalInstance: $uibModalInstance,
+      onTaskComplete: onTaskComplete,
+    });
+
+    var allLoadBalancerNames = {};
+
+    function finishInitialization() {
+      accountService.listAccounts('openstack').then(function (accounts) {
+        $scope.accounts = accounts;
+        $scope.state.accountsLoaded = true;
+
+        var accountNames = _.pluck($scope.accounts, 'name');
+        if (accountNames.length && accountNames.indexOf($scope.loadBalancer.account) === -1) {
+          $scope.loadBalancer.account = accountNames[0];
+        }
+
+        ctrl.accountUpdated();
+      });
+    }
+
+    function initializeLoadBalancerNames() {
+      loadBalancerReader.listLoadBalancers('openstack').then(function (loadBalancers) {
+        loadBalancers.forEach((loadBalancer) => {
+          let account = loadBalancer.account;
+          if (!allLoadBalancerNames[account]) {
+            allLoadBalancerNames[account] = {};
+          }
+          let region = loadBalancer.region;
+          if (!allLoadBalancerNames[account][region]) {
+            allLoadBalancerNames[account][region] = [];
+          }
+          allLoadBalancerNames[account][region].push(loadBalancer.name);
+        });
+
+        $scope.state.loadBalancerNamesLoaded = true;
+        updateLoadBalancerNames();
+      });
+    }
+
+    function updateLoadBalancerNames() {
+      $scope.existingLoadBalancerNames = _.flatten(_.map(allLoadBalancerNames[$scope.loadBalancer.account || ''] || []));
+    }
+
+    // Controller API
+    this.updateName = function() {
+      var loadBalancer = $scope.loadBalancer;
+      var loadBalancerName = [application.name, (loadBalancer.stack || ''), (loadBalancer.detail || '')].join('-');
+      loadBalancer.name = _.trimRight(loadBalancerName, '-');
+      return loadBalancer.name;
+    };
+
+    var currentRegionRequest = 0;
+    this.accountUpdated = function() {
+      updateLoadBalancerNames();
+
+      currentRegionRequest++;
+      var requestId = currentRegionRequest;
+      accountService.getRegionsForAccount($scope.loadBalancer.account).then(function(regions) {
+        if( requestId != currentRegionRequest ) {
+          return;
+        }
+
+        $scope.regions = _.map(regions, function(r) { return {label: r, value: r}; });
+      });
+    };
+
+    this.regionUpdated = function() {
+      //updating the filter triggers a refresh of the subnets
+      $scope.subnetFilter = {type: 'openstack', account: $scope.loadBalancer.account, region: $scope.loadBalancer.region};
+    };
+
+    this.onSubnetChanged = function() {
+      updateFloatingIps();
```

Comment:
I'll merge them. Might need it as a separate function later, but I'll break it out then.